From bd0e1b421dbbe815e43b6a67ca6d2985700e30be Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Thu, 3 May 2018 15:11:10 +0800
Subject: [PATCH 2/5] mediaextractor: add seccomp policy for x86_64

This is used to pass the following CTS test case on x86_64.

bug:69073312

Test: run cts -m CtsMediaTestCases and mediaextractor should not
crash due to minijail
Test: run cts -m CtsSecurityHostTestCases
-t android.security.cts.ProcessMustUseSeccompTest#testMediaextractorHasSeccompFilter
should pass
Test: run CtsMediaTestCases, CtsSecurityHostTestCases, CtsAppSecurityHostTestCases
and CtsMediaStressTestCases, ensure this patch doesn't introduce any regressions.

Change-Id: I440208195113a87baea8881b1fa0fe87a4910d95
Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 services/mediaextractor/Android.mk                 |  3 +-
 .../seccomp_policy/mediaextractor-x86_64.policy    | 62 ++++++++++++++++++++++
 2 files changed, 64 insertions(+), 1 deletion(-)
 create mode 100644 services/mediaextractor/seccomp_policy/mediaextractor-x86_64.policy

diff --git a/services/mediaextractor/Android.mk b/services/mediaextractor/Android.mk
index 1ebb7ff..57ac145 100644
--- a/services/mediaextractor/Android.mk
+++ b/services/mediaextractor/Android.mk
@@ -14,6 +14,7 @@ include $(CLEAR_VARS)
 LOCAL_REQUIRED_MODULES_arm := mediaextractor.policy
 LOCAL_REQUIRED_MODULES_arm64 := mediaextractor.policy
 LOCAL_REQUIRED_MODULES_x86 := mediaextractor.policy
+LOCAL_REQUIRED_MODULES_x86_64 := mediaextractor.policy
 LOCAL_SRC_FILES := main_extractorservice.cpp
 LOCAL_SHARED_LIBRARIES := libmedia libmediaextractorservice libbinder libutils \
     liblog libbase libicuuc libavservices_minijail
@@ -24,7 +25,7 @@ LOCAL_C_INCLUDES := frameworks/av/media/libmedia
 include $(BUILD_EXECUTABLE)
 
 # service seccomp filter
-ifeq ($(TARGET_ARCH), $(filter $(TARGET_ARCH), arm arm64 x86))
+ifeq ($(TARGET_ARCH), $(filter $(TARGET_ARCH), arm arm64 x86 x86_64))
 include $(CLEAR_VARS)
 LOCAL_MODULE := mediaextractor.policy
 LOCAL_MODULE_CLASS := ETC
diff --git a/services/mediaextractor/seccomp_policy/mediaextractor-x86_64.policy b/services/mediaextractor/seccomp_policy/mediaextractor-x86_64.policy
new file mode 100644
index 0000000..bcb948b
--- /dev/null
+++ b/services/mediaextractor/seccomp_policy/mediaextractor-x86_64.policy
@@ -0,0 +1,62 @@
+# Organized by frequency of systemcall - in descending order for
+# best performance.
+ioctl: 1
+futex: 1
+prctl: 1
+write: 1
+getpriority: 1
+close: 1
+dup: 1
+mmap: 1
+munmap: 1
+openat: 1
+mprotect: 1
+madvise: 1
+getuid: 1
+fstat: 1
+fstatfs: 1
+read: 1
+setpriority: 1
+sigaltstack: 1
+clone: 1
+lseek: 1
+mremap: 1
+newfstatat: 1
+faccessat: 1
+pread64: 1
+restart_syscall: 1
+exit: 1
+exit_group: 1
+rt_sigreturn: 1
+getrlimit: 1
+clock_gettime: 1
+writev: 1
+brk: 1
+sched_setscheduler: 1
+geteuid: 1
+getgid: 1
+getegid: 1
+getgroups: 1
+nanosleep: 1
+
+#for FileSource
+readlinkat: 1
+
+# for attaching to debuggerd on process crash
+tgkill: 1
+socket: arg0 == 1
+connect: 1
+recvmsg: 1
+rt_tgsigqueueinfo: 1
+writev: 1
+rt_sigaction: 1
+rt_sigprocmask: 1
+fcntl: 1
+getdents: 1
+pipe2: 1
+ppoll: 1
+
+# Required by AddressSanitizer
+gettid: 1
+sched_yield: 1
+getpid: 1
-- 
1.9.1

